skyplane/obj_store/s3_interface.py

```diff
@@ -79,6 +80,18 @@ def create_bucket(self, aws_region):
                 s3_client.create_bucket(Bucket=self.bucket_name, CreateBucketConfiguration={"LocationConstraint": aws_region})
 
     def delete_bucket(self):
+        # delete 1000 keys at a time
+        keys = []
+        for key in self.list_objects():
+            keys.append(key.key)
+            if len(keys) == 1000:
+                self.delete_objects(keys)
+                keys = []
+                print("Deleted 1000 keys")
+        self.delete_objects(keys)
```

Comment:
Seems that the `delete_objects` already has the logic of taking 1000 keys as the step to iterate.

Reply:
It's maybe better to use a generator here:
```python
    def delete_bucket(self):
        for batch in batch_generator(self.list_objects(), 1000):
            self.delete_objects([obj.key for obj in batch])
        # actually remove the bucket
```


There's already a convenience function that will batch data from a generator:
https://github.com/skyplane-project/skyplane/blob/da90f82602eda08f06ca32336ba10c7f1c7774a7/skyplane/api/transfer_job.py#L286-L299